src/app/login/login.component.ts

```diff
@@ -25,7 +25,7 @@ export class LoginComponent implements OnInit {
     if (!environment.test) {
       this.checkAdminExistence().subscribe((noAdmin) => {
         // false means there is admin
-        if (noAdmin) {
+        if (noAdmin && !this.checkAuthSession()) {

```

Comment:
this should check for is_logged in or has session. and redirect to dashboard. No need to check for admin exist if already logged in